topology/src/main/java/io/camunda/zeebe/topology/api/ScaleRequestTransformer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology.api;
+
+import io.atomix.cluster.MemberId;
+import io.camunda.zeebe.topology.changes.TopologyChangeCoordinator.TopologyChangeRequest;
+import io.camunda.zeebe.topology.state.ClusterTopology;
+import io.camunda.zeebe.topology.state.TopologyChangeOperation;
+import io.camunda.zeebe.util.Either;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ScaleRequestTransformer implements TopologyChangeRequest {
+
+  private final Set<MemberId> members;
+  private final ArrayList<TopologyChangeOperation> generatedOperations = new ArrayList<>();
+
+  public ScaleRequestTransformer(final Set<MemberId> members) {
+    this.members = members;
+  }
+
+  @Override
+  public Either<Exception, List<TopologyChangeOperation>> operations(
+      final ClusterTopology currentTopology) {
+    generatedOperations.clear();
+
+    // First add new members
+    return new AddMembersTransformer(members)
+        .operations(currentTopology)
+        .map(this::addToOperations)
```

Comment:
:x: I'm think we really shouldn't do this because the `map` is supposed to be an idempotent computation. I think in theory, the Java streams implementation is allowed to call this multiple times for the same element.

Reply:
Just a normal for-each for adding, reassigning and removing would be better I think.

Reply:
This is an Either#map though. Does that change your comment?

Reply:
:facepalm: Yes, then it's fine :+1: